agent/statechange/statechange.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package statechange
+
+const (
+	ContainerEvent = iota
```

Comment:
i was going back and forth on this as well. i decided to keep this with the interface since implementing this method will require you to decide which of those enums to return. but ya, i'm not sure which one is better.